### PR TITLE
Fixes for some Windows compiler warnings

### PR DESCRIPTION
**Pull request by vaeksare**
Implement some simple fixes to clean up windows build by fixing compiler warnings. Three main types of warnings were fixes:

1. GCC specific pragmas were changed to not be used on windows.
2. cmake flags that don't exist on windows were removed from windows build
3. Fix a macro that was defined multiple times on Windows.

### CODE REVIEW

File: aten/src/ATen/native/cpu/GridSamplerKernel.cpp [hunk 1]
```diff
@@ -627,7 +635,9 @@ struct ApplyGridSample<scalar_t, 2, GridSamplerInterpolation::Nearest, padding>
     integer_t gInp_offset_arr[iVec::size];
     i_gInp_offset.store(gInp_offset_arr);
 
-    #pragma unroll
+    #if (!defined _MSC_VER)
```
Comment (ezyang) [inline aten/src/ATen/native/cpu/GridSamplerKernel.cpp]: Wow! I didn't know you could write defined declarations this way.

File: aten/src/ATen/test/test_install/CMakeLists.txt [hunk 2]
```diff
@@ -3,6 +3,8 @@ find_package(ATen REQUIRED)
 include_directories(${ATEN_INCLUDE_DIR})
 
 # C++11
-set(CMAKE_CXX_FLAGS "--std=c++11 ${CMAKE_CXX_FLAGS}")
+if (not MSVC) 
+    set(CMAKE_CXX_FLAGS "--std=c++11 ${CMAKE_CXX_FLAGS}") 
+endif()
```
Comment (ezyang) [inline aten/src/ATen/test/test_install/CMakeLists.txt]: I'm... surprised this ever worked at all. (Don't we have to still tell MSVC to use C++11? Maybe it's default??)

File: aten/src/TH/THTensorApply.h [hunk 3]
```diff
@@ -247,10 +247,13 @@
 
 #ifdef _OPENMP
 
-#ifndef _WIN32
-#define PRAGMA(P) _Pragma(#P)
+#if (defined _WIN32)  
+// MSVC doesing support loop pragmas, but does support others. Create a new macro to account for those differences.
```
Comment (ezyang) [inline aten/src/TH/THTensorApply.h]: nit: doesn't

**Review by ezyang — APPROVED** [review]
Thank you very much. I think the prevailing convention in the codebase is to write `#if defined(_MSC_VER)` rather than the outer parentheses form, do you think you'd be willing switch them over to this form? If not, I can go ahead and do it for you. This patch is much appreciated, thank you.

**Comment by ezyang** [comment]
Travis failure is spurious, can be ignored

File: aten/src/ATen/native/cpu/GridSamplerKernel.cpp (hunk 1 above)
Comment (vaeksare) [reply aten/src/ATen/native/cpu/GridSamplerKernel.cpp]: I can change it to the other convention if you want (and haven't done so already).

File: aten/src/ATen/test/test_install/CMakeLists.txt (hunk 2 above)
Comment (vaeksare) [reply aten/src/ATen/test/test_install/CMakeLists.txt]: MSVC doesn't actually have a compiler option to use C++11 (see https://blogs.msdn.microsoft.com/vcblog/2016/06/07/standards-version-switches-in-the-compiler/). Previously the compiler was just throwing a warning and ignoring the flag. Given that everything seems to work fine, I guess this doesn't break anything?

File: aten/src/ATen/native/cpu/GridSamplerKernel.cpp (hunk 1 above)
Comment (ezyang) [reply aten/src/ATen/native/cpu/GridSamplerKernel.cpp]: Yes please!

File: aten/src/ATen/test/test_install/CMakeLists.txt (hunk 2 above)
Comment (ezyang) [reply aten/src/ATen/test/test_install/CMakeLists.txt]: SGTM

File: aten/src/ATen/native/cpu/GridSamplerKernel.cpp (hunk 1 above)
Comment (vaeksare) [reply aten/src/ATen/native/cpu/GridSamplerKernel.cpp]: Updated to use the standard convention.

**Pull request closed without merging**